controllers/workflow_controller.go

```diff
@@ -61,7 +61,7 @@ type WorkflowReconciler struct {
 // checkDriverStatus returns true if all registered drivers for the current state completed successfully
 func checkDriverStatus(instance *dwsv1alpha1.Workflow) (bool, error) {
 	for _, d := range instance.Status.Drivers {
-		if d.WatchState == instance.Spec.DesiredState {
+		if d.WatchState == instance.Status.State {
```

Comment:
What are the symptoms of this bug?
This should be mentioned in the  commit description.

Reply:
It was just something I noticed while reading the code.